packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -90,6 +90,13 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   /// Return `null` if nothing is selected.
   SelectedContent? getSelectedContent();
 
+  /// Gets the [SelectedContentRange] representing the selected range in this object.
+  ///
+  /// When nothing is selected, subclasses should return a [SelectedContentRange.empty],
+  /// with a [SelectedContentRange.contentLength] that represents the length
+  /// of the content under this [SelectionHandler].
+  SelectedContentRange getSelection();
```

Comment:
I thought about this, we would have to change the contract of `getSelectedContent` since it returns `null` when there is nothing selected, while the new API returns an `empty` representation of the selection. Also we would be exposing the `start` and `end` offsets through `SelectionArea.onSelectionChanged`. I think exposing those offsets through `SelectionArea.onSelectionChanged` might make them difficult to consume depending on the complexity of the `Selectable` tree, `SelectionListener` solves for this by allowing the user to get offsets for a specific subtree. I do think these APIs are similar/related but not exactly sure how to merge them unless we are okay with exposing the start and end offsets at the `SelectionArea` level. What do you think about this?